symfony/plugins/orangehrmPimPlugin/Dao/EmployeeDao.php

```diff
@@ -0,0 +1,118 @@
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Pim\Dao;
+
+use Exception;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\ORM\Doctrine;
+use OrangeHRM\ORM\Paginator;
+use OrangeHRM\Pim\Dto\EmployeeSearchParamHolder;
+
+class EmployeeDao
+{
+    /**
+     * @param EmployeeSearchParamHolder $employeeSearchParamHolder
+     * @return array
+     * @throws DaoException
+     */
+    public function getEmployeeList(EmployeeSearchParamHolder $employeeSearchParamHolder): array
+    {
+        try {
+            $paginator = $this->getEmployeeListPaginator($employeeSearchParamHolder);
+            return $paginator->getQuery()->execute();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param EmployeeSearchParamHolder $employeeSearchParamHolder
+     * @return int
+     * @throws DaoException
+     */
+    public function getEmployeeCount(EmployeeSearchParamHolder $employeeSearchParamHolder): int
+    {
+        try {
+            $paginator = $this->getEmployeeListPaginator($employeeSearchParamHolder);
+            return $paginator->count();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param EmployeeSearchParamHolder $employeeSearchParamHolder
+     * @return Paginator
+     */
+    public function getEmployeeListPaginator(EmployeeSearchParamHolder $employeeSearchParamHolder): Paginator
+    {
+        $q = Doctrine::getEntityManager()->getRepository(Employee::class)->createQueryBuilder('e');
+        if (!is_null($employeeSearchParamHolder->getSortField())) {
+            $q->addOrderBy($employeeSearchParamHolder->getSortField(), $employeeSearchParamHolder->getSortOrder());
+        }
+        if (!empty($employeeSearchParamHolder->getLimit())) {
+            $q->setFirstResult($employeeSearchParamHolder->getOffset())
+                ->setMaxResults($employeeSearchParamHolder->getLimit());
+        }
+        if (!$employeeSearchParamHolder->isIncludeTerminated()) {
+            $q->andWhere($q->expr()->isNull('e.employeeTerminationRecord'));
+        }
+        if (!is_null($employeeSearchParamHolder->getName())) {
+            $q->andWhere(
+                $q->expr()->orX(
+                    $q->expr()->like('e.firstName', ':name'),
+                    $q->expr()->like('e.lastName', ':name'),
+                    $q->expr()->like('e.middleName', ':name'),
+                )
+            );
+            $q->setParameter('name', '%' . $employeeSearchParamHolder->getName() . '%');
```

Comment:
Help wanted: @samanthajayasinghe do you have any idea how to handle LIKE query with doctrine.
some refs: [1](https://stackoverflow.com/questions/3755718/doctrine2-dql-use-setparameter-with-wildcard-when-doing-a-like-comparison), [2](https://stackoverflow.com/questions/2843009/how-to-escape-like-var-with-doctrine/7893670#7893670), [3](https://www.doctrine-project.org/projects/doctrine-orm/en/2.8/reference/query-builder.html#the-expr-class)